source/MaterialXView/Handlers/DefaultGeometryHandler.cpp

```diff
@@ -29,9 +41,264 @@ void DefaultGeometryHandler::setIdentifier(const std::string identifier)
+                        }
+                    }
+                }
+            }
+        }
+
+        dump.close();
+        objfile.close();
+
+        // Set bounds
+        _minimumBounds[0] = minPos[0];
+        _minimumBounds[1] = minPos[1];
+        _minimumBounds[2] = minPos[2];
+        _maximumBounds[0] = maxPos[0];
+        _maximumBounds[1] = maxPos[1];
+        _maximumBounds[2] = maxPos[2];
+
+        // Organize data to get triangles for positions 
+        for (unsigned int i = 0; i < pidx.size() ; i++)
+        {
+            unsigned int vertexIndex = 3 * pidx[i];
+            _positionData.push_back(pos[vertexIndex]);
+            _positionData.push_back(pos[vertexIndex + 1]);
+            _positionData.push_back(pos[vertexIndex + 2]);
+        }
+
+        // Organize data to get triangles for texture coordinates 
+        for (unsigned int i = 0; i < uvidx.size(); i++)
+        {
+            unsigned int vertexIndex = 2 * uvidx[i];
+            _texcoordData[0].push_back(uv[vertexIndex]);
+            _texcoordData[0].push_back(uv[vertexIndex + 1]);
+            
+            _texcoordData[1].push_back(uv[vertexIndex + 1]);
+            _texcoordData[1].push_back(uv[vertexIndex]);
+
+            // Fake some colors
+            _colorData[0].push_back(uv[vertexIndex]);
+            _colorData[0].push_back(uv[vertexIndex] + 1);
+            _colorData[0].push_back(1.0f);
+            _colorData[0].push_back(1.0f);
+            
+            _colorData[1].push_back(1.0f);
+            _colorData[1].push_back(uv[vertexIndex] + 1);
+            _colorData[1].push_back(uv[vertexIndex]);
+            _colorData[1].push_back(1.0f);
+        }
+
+        // Organize data to get triangles for normals 
+        for (unsigned int i = 0; i < nidx.size(); i++)
+        {
+            unsigned int vertexIndex = 3 * nidx[i];
+            _normalData.push_back(norm[vertexIndex]);
+            _normalData.push_back(norm[vertexIndex + 1]);
+            _normalData.push_back(norm[vertexIndex + 2]);
+
+            // Fake some tangent, bitangent data
+            _tangentData[0].push_back(norm[vertexIndex]);
+            _tangentData[0].push_back(norm[vertexIndex + 1]);
+            _tangentData[0].push_back(norm[vertexIndex + 2]);
+
+            _bitangentData[0].push_back(norm[vertexIndex]);
+            _bitangentData[0].push_back(norm[vertexIndex + 1]);
+            _bitangentData[0].push_back(norm[vertexIndex + 2]);
+        }
+
+        // Set up flattened indexing
+        if (_positionData.size() && pidx.size())
+        {
+            _indexing.resize(pidx.size());
+            std::iota(_indexing.begin(), _indexing.end(), 0);
+        }
+    }
 }
 
-GeometryHandler::IndexBuffer& DefaultGeometryHandler::getIndexing(size_t &bufferSize)
+const MaterialX::Vector3& DefaultGeometryHandler::getMinimumBounds()
+{
+    if (_identifier == SCREEN_ALIGNED_QUAD)
```

Comment:
For all the special handling of SCREEN_ALIGNED_QUAD, why not just create the data for the quad at geometry load time (same place that loads the obj file). And then all the access metods can just return all geometry data in the same way without special handling.

Reply:
Good idea, I'll create a plane and make the camera fit.